src/notificationarea.cpp

```diff
@@ -168,4 +169,60 @@ void NotificationArea::setSettings(const QString &placement, int width, int spac
 
     m_layout->setUnattendedMaxNum(unattendedMaxNum);
     m_layout->setBlackList(blackList);
+
+        // layer-shell code
+    bool underWayland = QGuiApplication::platformName() == QStringLiteral("wayland");
+    QMargins margins(m_spacing, m_spacing, m_spacing, m_spacing);
+    if (underWayland)
+       LayerShellQt::Shell::useLayerShell();
+    if (underWayland)
+    {
+        winId();
+        if(QWindow* win = windowHandle())
+        {
+            if(LayerShellQt::Window* layershell = LayerShellQt::Window::get(win))
+            {
+                layershell->setLayer(LayerShellQt::Window::Layer::LayerOverlay);
+                layershell->setKeyboardInteractivity(LayerShellQt::Window::KeyboardInteractivityNone);
+                layershell->setMargins(margins);
+               // layershell->setScope(scope);
+                if (QL1S("top-left") == m_placement)
+                {
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorTop|LayerShellQt::Window::AnchorLeft};
+                    layershell->setAnchors(anchors);
+                }
+                if (QL1S("top-center") == m_placement)
+                {
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorTop};
+                    layershell->setAnchors(anchors);
+                } else if (QL1S("top-right") == m_placement)
+                {
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorTop|                LayerShellQt::Window::AnchorRight};
+                    layershell->setAnchors(anchors);
+                } else if (QL1S("center-left") == m_placement)
+                {
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorLeft};
+                    layershell->setAnchors(anchors);
+                } else if (QL1S("center-right") == m_placement)
+                {
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorRight};
+                    layershell->setAnchors(anchors);
+                } else if (QL1S("bottom-left") == m_placement)
+                {
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorLeft|                LayerShellQt::Window::AnchorBottom};
+                    layershell->setAnchors(anchors);
+                } else if (QL1S("bottom-center") == m_placement)
+                {
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorBottom};
+                    layershell->setAnchors(anchors);
+                } else if (QL1S("bottom-right") == m_placement)
+                {
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorBottom|                LayerShellQt::Window::AnchorRight};
+                    layershell->setAnchors(anchors);
+                }
```

Comment:
```c++
               LayerShellQt::Window::Anchors anchors;
                if (m_placement == QL1S("top-left"))
                    anchors = {LayerShellQt::Window::AnchorTop|LayerShellQt::Window::AnchorLeft};
                else if (m_placement == QL1S("top-center"))
                    anchors = {LayerShellQt::Window::AnchorTop};
                else if (m_placement == QL1S("top-right"))
                    anchors = {LayerShellQt::Window::AnchorTop | LayerShellQt::Window::AnchorRight};
                // the other conditions come here......
                layershell->setAnchors(anchors);
```

Reply:
This gives meduring compiling several times:```

/tmp/lxqt-notificationd/src/notificationarea.cpp:191:51: warning: variable ‘anchors’ set but not used [-Wunused-but-set-variable]
  191 |                     LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorTop|LayerShellQt::Window::AnchorLeft};

```

and the last one, bottom right, is only working.

Reply:
Perhaps because you repeated `LayerShellQt::Window::Anchors`.